app/controllers/DataSetController.scala

```diff
@@ -72,6 +75,21 @@ class DataSetController @Inject()(val messagesApi: MessagesApi) extends Controll
     }
   }
 
+  def addForeignDataStoreAndDataSet(url: String, dataStoreName: String, dataSetName: String) = SecuredAction.async { implicit request =>
+    for {
+      dataStoreBox <- DataStoreDAO.findOneByName(dataStoreName).reverse.futureBox
+      _ <- dataStoreBox match {
+        case Full(_) => DataSetService.addForeignDataStore(dataStoreName, url)
```

Comment:
Does this not give a match error if the box is not full? I’d have thought we also need `case _ => Fox.successful(())`

Reply:
Also, I’d suggest not storing the Fox.reverse, but rather the normal Box[dataStore], and then insert on case Empty. Seems easier to understand